libs/datasets/dataset_base.py

```diff
@@ -48,3 +43,14 @@ def to_csv(self, path: pathlib.Path):
         if self.provenance is not None:
             provenance_path = str(path).replace(".csv", "-provenance.csv")
             self.provenance.sort_index().to_csv(provenance_path)
+
+    def indexed_df(self) -> pd.DataFrame:
```

Comment:
Could you add a docstring here? 

Also, I might propose making this a property:
```
@property
def indexed_data(self):
    return ...

```

Might give a bit more symmetry accessing: 
`self.data` and `self.indexed_data`

Reply:
added docstring
Symmetry is good but I also like making cheap reads simple object attributes and more expensive ones methods.

Reply:
how about `indexed_data` rather than `indexed_df`? We don't really refer to df anywhere - so data feels a bit more consistent

Reply:
Done in 502a9ec